pkg/retry/percentage_limit.go

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package retry
+
+import (
+	"context"
+
+	"github.com/cloudwego/kitex/pkg/circuitbreak"
+)
+
+type typeRetryCallTimes int
+
+var keyRetryCallTimes typeRetryCallTimes
+
+// ctxWithCallTimes should be called to set the key, regardless of whether it's a retry request, for faster unwrap
+func ctxWithCallTimes(ctx context.Context, callTimes int32) context.Context {
+	return context.WithValue(ctx, keyRetryCallTimes, callTimes)
+}
+
+// ctxGetRetryCallTimes returns the call time; 1 means first call (not retry)
+func ctxGetRetryCallTimes(ctx context.Context) int {
+	v := ctx.Value(keyRetryCallTimes)
+	if v == nil {
+		return 0
+	}
+	if callTimes, ok := v.(int32); ok {
+		return int(callTimes)
+	}
+	return 0
+}
+
+// TreatRetryAsError regards a retry request as failure
+// It's used as CBSuite.ServiceControl().GetErrorType for limiting retry requests
+func TreatRetryAsError(ctx context.Context, req, rsp interface{}, err error) circuitbreak.ErrorType {
+	if ctxGetRetryCallTimes(ctx) > 1 {
```

Comment:
I think `IsLocalRetryRequest` is enough here

Reply:
As discussed, the RetryTag is set in rpcCall and is not visible here. And it's solved in a much better way, so we don't need this function anymore.